torch/fx/examples/custom_tracer.py

```diff
@@ -0,0 +1,115 @@
+import torch
+from torch.fx import symbolic_trace, Tracer, Graph, GraphModule, Node
+from typing import Any, Callable, Dict, Optional, Tuple, Union
+
+"""
+How to Create and Use Custom Tracers
+
+`Tracer`--the class that implements the symbolic tracing functionality
+of `torch.fx.symbolic_trace`--can be subclassed to override various
+behaviors of the tracing process. In this tutorial, we'll demonstrate
+how to customize the symbolic tracing process using some handwritten
+Tracers. Each example will show that, by simply overriding a few methods
+in the `Tracer` class, you can completely alter the Graph produced by
+symbolic tracing. For a complete description of the methods that
+can be changed, refer to the docstrings of the methods in the Tracer
+class.
+
+If you want a real-world example of a custom tracer, check out FX's AST
+Rewriter in `rewriter.py`. `RewritingTracer` inherits from Tracer but
+overrides the `trace` function so that we can rewrite all calls to
+`assert` to the more FX-friendly `torch.assert`.
+
+Note that a call to `symbolic_trace(m)` is equivalent to
+`Tracer().trace(m).graph`. (`Tracer` is the default implementation of
+Tracer as defined in `symbolic_trace.py`.)
```

Comment:
Same comment as below, it's the other way around. `symbolic_trace` returns a `GraphModule` but `Tracer.trace` returns a `Graph`